packages/purpose-process/src/services/readModelService.ts

```diff
@@ -123,6 +135,123 @@ export function readModelServiceBuilder(
         },
       } satisfies ReadModelFilter<Purpose>);
     },
+    async getPurposes(
+      filters: GetPurposesFilters,
+      { offset, limit }: { offset: number; limit: number }
+    ): Promise<ListResult<Purpose>> {
+      const {
+        name,
+        eservicesIds,
+        consumersIds,
+        producersIds,
+        states,
+        excludeDraft,
+      } = filters;
+
+      const nameFilter: ReadModelFilter<Purpose> = name
+        ? {
+            "data.title": {
+              $regex: ReadModelRepository.escapeRegExp(name),
+              $options: "i",
+            },
+          }
+        : {};
+
+      const eservicesIdsFilter: ReadModelFilter<Purpose> =
+        ReadModelRepository.arrayToFilter(eservicesIds, {
+          "data.eserviceId": { $in: eservicesIds },
+        });
+
+      const consumersIdsFilter: ReadModelFilter<Purpose> =
+        ReadModelRepository.arrayToFilter(consumersIds, {
+          "data.consumerId": { $in: consumersIds },
+        });
+
+      const versionStateFilter: ReadModelFilter<Purpose> =
+        ReadModelRepository.arrayToFilter(states, {
+          "data.versions.state": { $in: states },
+        });
+
+      const draftFilter: ReadModelFilter<Purpose> = excludeDraft
+        ? {
+            $nor: [
+              { "data.versions": { $size: 0 } },
+              {
+                $and: [
+                  { "data.versions": { $size: 1 } },
+                  {
+                    "data.versions.state": {
+                      $eq: purposeVersionState.draft,
+                    },
+                  },
+                ],
+              },
+            ],
+          }
+        : {};
+
+      const eserviceIds =
+        producersIds.length > 0
+          ? await eservices
+              .find({ "data.producerId": { $in: producersIds } })
+              .toArray()
+              .then((results) =>
+                results.map((eservice) => eservice.data.id.toString())
+              )
+          : [];
+
+      const producerIdsFilter: ReadModelFilter<Purpose> =
+        ReadModelRepository.arrayToFilter(eserviceIds, {
+          "data.eserviceId": { $in: eserviceIds },
+        });
+
+      const aggregationPipeline = [
```

Comment:
the filter by archived purposes is missing

Reply:
Thanks, I completely missed that part, sorry. Updated